### PR TITLE
Add encryption keys to site config add keyring.Ring type to load & configure keys from config

### PR DESCRIPTION
**Pull request by arussellsaw**
This PR does the following:

* adds encryption key config to site config
* renames `keyring` 👉  `encryption`
* adds subpackage `encryption/keyring` for loading keys from site config
* adds `NoopKey` so we can wire stuff up without actually encrypting
* loads cloud KMS keys from config

cc @mrnugget this is the work i was talking about earlier.

### CODE REVIEW

File: internal/encryption/key.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package keyring
+package encryption
```
Comment (tsenart) [inline internal/encryption/key.go:1]: How about `crypto`?

**Review by tsenart — APPROVED** [review]

File: internal/encryption/key.go (hunk 1 above)
Comment (ryanslade) [reply internal/encryption/key.go:1]: That would conflict with the standard lib

File: internal/encryption/keyring/ring.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	"context"
+	"fmt"
+
+	"github.com/sourcegraph/sourcegraph/internal/encryption/cloudkms"
+
+	"github.com/sourcegraph/sourcegraph/internal/encryption"
+	"github.com/sourcegraph/sourcegraph/schema"
+)
+
+// NewRing  creates a encryption.Ring containing all the keys configured in site config
```
Comment (ryanslade) [inline internal/encryption/keyring/ring.go:13]:
```suggestion
// NewRing creates a keyring.Ring containing all the keys configured in site config
```

File: internal/encryption/noop.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import "context"
+
+var _ Key = &NoopKey{}
+
+type NoopKey struct{}
+
+func (k *NoopKey) Encrypt(ctx context.Context, plaintext []byte) ([]byte, error) {
+	return plaintext, nil
+}
+func (k *NoopKey) Decrypt(ctx context.Context, ciphertext []byte) (*Secret, error) {
```
Comment (ryanslade) [inline internal/encryption/noop.go:12]: nit: Missing space between functions

**Review by ryanslade — APPROVED** [review]

**Pull request merged**